tests/Unit/Model/Export/ProductTest.php

```diff
@@ -155,6 +155,83 @@ public function testEmptyAttributeGroups($productType)
         $this->assertEmpty($exportModel->getAttributeGroups());
     }
 
+    /**
+     * @param array     $imageData
+     * @param string    $smallImagePath
+     *
+     * @covers ::setImages
+     * @dataProvider imageProvider
+     */
+    public function testImageSortOrder($imageData, $smallImagePath)
+    {
+        $productStub = $this->getProductDouble();
+        $productStub->method('getMediaGalleryImages')
+            ->will($this->returnValue($this->getTestCollection($imageData)));
+
+        $productStub->method('getData')
+            ->will($this->returnValueMap([['small_image', null, $smallImagePath]]));
+
+        /** @var \Shopgate\Export\Model\Export\Product $exportModel */
+        $exportModel = $this->objectManager->getObject(
+            'Shopgate\Export\Model\Export\Product'
+        );
+
+        $exportModel->setItem($productStub)->setImages();
+
+        $images = array_reverse($exportModel->getImages());
+        $firstImage = array_pop($images);
+
+        $this->assertEquals($smallImagePath, $firstImage->getUrl());
+        $this->assertEquals(1, $firstImage->getIsCover());
+    }
+
+    /**
+     * @return array
+     */
+    public function imageProvider()
+    {
+        return [
+            'First product set as small image' => [[$this->createImageObject('1.jpg', '1'), $this->createImageObject('2.jpg', '3')], '1.jpg'],
+            'Second product set as small image' => [[$this->createImageObject('1.jpg', '1'), $this->createImageObject('2.jpg', '14')], '2.jpg']
+        ];
+    }
+
+    /**
+     * @param string    $url
+     * @param int       $position
+     *
+     * @return \Magento\Framework\DataObject
+     */
+    private function createImageObject($url, $position)
+    {
+        return new \Magento\Framework\DataObject([
+            'id'       => rand(0, 10000),
```

Comment:
mt_rand is recommended instead
```suggestion
            'id'       => mt_rand(0, 10000),
```